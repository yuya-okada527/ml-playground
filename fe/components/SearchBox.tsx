import React, { ChangeEvent } from "react";
import {
  Button,
  createStyles,
  makeStyles,
  TextField,
  Theme,
  Typography,
  Grid,
} from "@material-ui/core";

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    searchButton: {
      marginTop: theme.spacing(1.5),
      marginLeft: theme.spacing(3),
    },
  })
);

type SearchBoxProps = {
  searchTerm: string;
  handleSearchTermChange: (event: ChangeEvent<HTMLInputElement>) => void;
  handleSearchButtonClick: () => void;
};

const SearchBox = ({
  searchTerm,
  handleSearchTermChange,
  handleSearchButtonClick,
}: SearchBoxProps) => {
  const classes = useStyles();
  return (
    <>
      <Typography variant="h6" component="h2">
        Search Your Favorite Movies!!
      </Typography>
      <Grid container>
        <Grid item xs={10}>
          <TextField
            id="search-movie"
            label="Search Movies!!"
            type="search"
            fullWidth
            value={searchTerm}
            onChange={handleSearchTermChange}
          />
        </Grid>
        <Grid item xs={2}>
          <Button
            variant="contained"
            color="primary"
            className={classes.searchButton}
            onClick={handleSearchButtonClick}
          >
            Search
          </Button>
        </Grid>
      </Grid>
    </>
  );
};

export default SearchBox;
