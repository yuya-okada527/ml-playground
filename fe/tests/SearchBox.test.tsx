import React from "react";
import { shallow } from "enzyme";
import SearchBox from "../src/components/SearchBox";

describe("SearchBox", () => {
  it("Snapshot Test", () => {
    const searchBox = shallow(
      <SearchBox
        searchTerm="searchTerm"
        handleSearchTermChange={jest.fn()}
        handleSearchButtonClick={jest.fn()}
      />
    );
    expect(searchBox).toMatchSnapshot();
  });
});
