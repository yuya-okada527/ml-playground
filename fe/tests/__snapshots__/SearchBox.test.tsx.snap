// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox Snapshot Test 1`] = `
<Fragment>
  <WithStyles(ForwardRef(Typography))
    component="h2"
    variant="h6"
  >
    Search Your Favorite Movies!!
  </WithStyles(ForwardRef(Typography))>
  <WithStyles(ForwardRef(Grid))
    container={true}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={10}
    >
      <WithStyles(ForwardRef(TextField))
        fullWidth={true}
        id="search-movie"
        label="Search Movies!!"
        onChange={[MockFunction]}
        type="search"
        value="searchTerm"
      />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={2}
    >
      <WithStyles(ForwardRef(Button))
        className="makeStyles-searchButton-1"
        color="primary"
        onClick={[MockFunction]}
        variant="contained"
      >
        Search
      </WithStyles(ForwardRef(Button))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</Fragment>
`;
